File: 프로그래머스/3/12979. 기지국 설치/기지국 설치.py
def solution(n, stations, w):
    answer = 0
    cover = w*2 +1
    start = 1
    for station in stations:
        lt, rt = station-w, station+w
    
        if start < lt:
            length = lt - start
            answer += (length+cover-1)//cover
            
        start = rt+1
        
    if start <= n:
        length = n-start +1
        answer += (length+cover-1)//cover
        
    
    

    return answer

File: 프로그래머스/3/12979. 기지국 설치/기지국 설치.java
import java.util.*;
class Solution {
    public int solution(int n, int[] stations, int w) {
        int answer = 0;
        int idx = 1;
        int wide = w*2+1;
        for(int station: stations){
            int lt = station-w;
            int rt = station+w;
            int uncover = lt-idx;
            if(uncover>0){
                answer += (uncover+wide-1)/wide;
            }
            idx = rt+1;
        }
        if(idx <= n){
            int uncover = n - idx +1;
            answer += (uncover+wide-1)/wide;
        }

        return answer;
    }
}